### ISSUE TITLE
cloudformation.WaitUntilStackCreateComplete waits forever

### ISSUE THREAD

**Issue by michaeleekk** [aws/aws-sdk-go-v2#92]
### Version of AWS SDK for Go?

```
{
	"checksumSHA1": "VsyHSneU5fMUAaQDQ1BAqZcvrG0=",
	"path": "github.com/aws/aws-sdk-go-v2",
	"revision": "01b01bc24609090862fbae454a86704b75a0f3b3",
	"revisionTime": "2018-01-11T17:45:17Z"
},
```

### Version of Go (`go version`)?

1.9.2

### What issue did you see?

I was just tried to upgrade from v1 to v2. I'm using cloudformation API. I could successfully created a stack with `CreateStackRequest` and got the stack id. But then when I used `WaitUntilStackCreateComplete`, it never returned.

### Steps to reproduce

```
	// create stack
	req := client.CreateStackRequest(&input)
	output, err := req.Send()
	if err != nil {
		return
	}
	fmt.Println(output)

	// wait the stack to be created
	describeInput := cf.DescribeStacksInput{
		StackName: output.StackId,
	}
	fmt.Printf("Waiting until the stack is created...")
	if err = client.WaitUntilStackCreateComplete(&describeInput); err != nil {
		fmt.Println(err)
		deleteStack(stackId)
		return
	}
	fmt.Println("Done.")
```

This is a snippet of my code. I never saw the `Done` printing out.

If you have have an runnable example, please include it.

**Comment by Doug-AWS** [comment]
I just wrote an example for the developer guide, but don't have the text for the topic written yet. I tested it against the template S3_Website_Bucket_With_Retain_On_Delete.template, which I found at https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/sample-templates-services-us-west-2.html#w2ab2c23c48c13c33. 

https://github.com/awsdocs/aws-doc-sdk-examples/blob/master/go/example_code/cloudformation/CfnCreateStack.go

**Comment by jasdel** [comment]
Thanks for letting us know about this @michaeleekk. Do you know if this waiting method works correctly with the v1 sdk?

Also if you could enable logging by setting `LogLevel` in the service client or Config before the client is created. To `aws.LogDebugWithHTTPBody`. it will print the request and response to stdout. This will help debugging this issue a lot.

**Comment by michaeleekk** [comment]
Thanks @doug-AWS, but sadly I think that’s what I did.

@jasdel I can confirm that the exact same program works for v1. But the log level check, I think I could only do it next Monday. Will get back to this issue when I have more info.

**Comment by robertojrojas** [comment]
I'm running into a similar issue with the method **ec2Svc.WaitUntilInstanceStatusOk(...)**. It works well on v1, but on v2 waits until it the error `ResourceNotReady: exceeded wait attempts` is returned. 
Here is a sample log

[aws-sdk-go-v2-issue-92.txt](https://github.com/aws/aws-sdk-go-v2/files/1640162/aws-sdk-go-v2-issue-92.txt)

**Comment by jasdel** [comment]
Thanks for the sample code and log. I think i've identified the issue with the waiters. The Waiters are failing because of the typed enumerations adding are causing the reflection based type comparison to fail because a typed aliased string does not DeepEqual with a regular string.

**Comment by jasdel** [comment]
I've put out PR #102 which corrects the SDK's behavior and no longer waits until a waiter expires.

**Comment by jasdel** [comment]
Thanks for brining this issue to our attention. we've merged in #102 which corrects this issue. Let us know if you run into any additional issues, or have feedback.
